src/sidebar/services/groups.js

```diff
@@ -348,25 +323,103 @@ export default function groups(
       directLinkedGroupId
     );
 
+    addGroupsToStore(groups, [
+      directLinkedAnnotationGroupId,
+      directLinkedGroupId,
+      store.getDefault('focusedGroup'),
+    ]);
+
+    return groups;
+  }
+
+  /**
+   * Load the specific groups configured by the annotation service.
+   *
+   * @param {string[]} groupIds
+   */
+  async function loadServiceSpecifiedGroups(groupIds) {
+    let error;
+    const tryFetchGroup = async id => {
+      try {
+        return await fetchGroup(id);
+      } catch (e) {
+        error = e;
+        return null;
+      }
+    };
+
+    const groups = (
+      await Promise.all(groupIds.map(id => tryFetchGroup(id)))
+    ).filter(g => g !== null);
+
+    addGroupsToStore(groups);
+
+    if (error) {
+      toastMessenger.error(`Unable to fetch groups: ${error.message}`, {
```

Comment:
This error would always pertain to a specific group (singular), yes, although, yes, it does indicate a failure in fetching the whole set of groups...

I'm wondering about this error message. If a day-to-day user saw this, would it be useful in any way? Confusing? Do our users know what a "group" is? Just musing...

Reply:
They might not understand exactly what the error means, but I think that's OK. The main goal is to clearly indicate that there was a problem and private some message unique to the problem but not too cryptic that they can quote back to us in any support requests.